### PR TITLE
Broken fallback for uploaded report name

### PR DESCRIPTION
**Pull request by dcblundell**
This pull request improves the handling and validation of folder-style file uploads in the backend, ensuring that uploaded files from different browsers (Chromium, Firefox, Safari) are consistently placed in the correct directory structure and that invalid mixed-folder uploads are rejected. It introduces a new helper for resolving the destination folder name, strengthens validation to prevent silent misplacement of files, and adds comprehensive tests to cover these behaviors.

**Folder name resolution and validation improvements:**

- Added `resolve_parent_folder_name` to consistently determine the destination folder for uploads, preferring an explicit form field when present (Safari) and falling back to filename inference (Chromium/Firefox). This centralizes logic and prevents files from being saved at the root or in incorrect directories. [[1]](diffhunk://#diff-6018bac330806062feec2dc1382f1bc01187e80ad9021456d5bd4719a9e2edf3L49-R108) [[2]](diffhunk://#diff-34e6863077d645b69bcb1a182649fa531a6fc979eff410b026cdb9158d70405aL43-R44) [[3]](diffhunk://#diff-34e6863077d645b69bcb1a182649fa531a6fc979eff410b026cdb9158d70405aL1275-R1280) [[4]](diffhunk://#diff-34e6863077d645b69bcb1a182649fa531a6fc979eff410b026cdb9158d70405aL1342-R1347)
- Updated `validate_files` to reject uploads that span multiple inferred parent folders, preventing cases where files from different folders would be incorrectly nested under a single report folder. [[1]](diffhunk://#diff-6018bac330806062feec2dc1382f1bc01187e80ad9021456d5bd4719a9e2edf3R24-R35) [[2]](diffhunk://#diff-6018bac330806062feec2dc1382f1bc01187e80ad9021456d5bd4719a9e2edf3R47-R61)

**File path construction fixes:**

- Modified `construct_dest_path` to deduplicate the leading folder segment in Chromium-style uploads, avoiding double-prefixing (e.g., `report/report/file`). This ensures files always land under the correct directory regardless of browser.

**Testing enhancements:**

- Added comprehensive tests for `resolve_parent_folder_name`, `validate_files`, and `construct_dest_path` covering all major browser behaviors and edge cases (e.g., empty fields, mixed-folder uploads, path traversal posture). [[1]](diffhunk://#diff-032e5c5e02f9e3f85e45226c88e8a04dbf043977ce7077f679a6c222995a25caR105-R180) [[2]](diffhunk://#diff-032e5c5e02f9e3f85e45226c88e8a04dbf043977ce7077f679a6c222995a25caR208-R297)
- Added end-to-end regression tests for profiler and performance uploads to verify that files land in the correct folder and that mixed-folder uploads are rejected.

These changes ensure robust, predictable handling of folder-style uploads across browsers and prevent subtle bugs from misplacing or nesting files incorrectly.

### CODE REVIEW

File: backend/ttnn_visualizer/file_uploads.py [hunk 1]
```diff
@@ -168,7 +196,16 @@ def construct_dest_path(file, target_directory, folder_name):
         # depend on that. Path-traversal hardening for the folder branch is a
         # separate, broader follow-up — see PR_REVIEW_TRIAGE_2.md §1.J.
         prefixed_folder_name = f"{prefix}{folder_name}"
-        dest_path = Path(target_directory) / prefixed_folder_name / str(file.filename)
+        # Chromium-based browsers send each file's relative path as the
+        # multipart filename (e.g. `report/db.sqlite`), while Safari sends just
+        # the basename and the destination folder name as a separate form
+        # field. Strip a duplicate leading segment so we land at
```
[the remark this answers is not in the document]
Comment (dcblundell) [reply backend/ttnn_visualizer/file_uploads.py:226]: The broader path-traversal sweep on the folder branch is the explicit `PR_REVIEW_TRIAGE_2.md §1.J` follow-up and predates this PR — keeping it out of scope here so the bug fix can ship narrow.

In the meantime this PR adds `test_construct_dest_path_folder_branch_traversal_posture_pinned`, which pins the current invariants:

1. **Hard:** dest must stay inside `target_directory` (any future change that breaks this is a real regression).
2. **Documented gap:** a `..` segment after the leading folder name CAN escape the per-report directory.

When the §1.J commit lands it should flip assertion (2) to "lands under `<target>/<folder>/`" without touching (1), so the pinned posture acts as a safety net both directions.

For the specific "absolute right-hand side" case (e.g. `myreport//etc/passwd`): `validate_files` runs `Path(file.filename)` first, which normalises the double slash and rejects the file because `parents == 3`, so that variant is already blocked at validation.

Resolving on the strength of the pinned posture test + tracked follow-up.

File: backend/ttnn_visualizer/file_uploads.py [hunk 2]
```diff
@@ -46,14 +46,42 @@ def validate_files(files, required_files, pattern=None, folder_name=None):
...
+
+    * As an explicit ``folderName`` form field (Safari, where the multipart
+      filename is just the basename and the relative path is lost), or
+    * Implicitly, as the leading segment of each file's relative path
+      (Chromium / Firefox preserve `webkitRelativePath` in the multipart
+      filename).
+
+    Centralising this resolution keeps the two upload handlers aligned and
+    avoids the easy mistake of passing the raw ``folder_name`` form field
+    (which is ``None`` for non-Safari clients) straight through to
```
[the remark this answers is not in the document]
Comment (dcblundell) [reply backend/ttnn_visualizer/file_uploads.py:100]: Updated.

**Review by smountenay-tt — APPROVED** [review]
